test/jdk/java/awt/event/MouseWheelEvent/WheelModifier/WheelModifier.java

```diff
@@ -117,33 +131,47 @@ void run() throws Exception {
 
         r.mouseMove(sLoc.x + bSize.width / 2, sLoc.y + bSize.height / 2);
         r.mousePress(MouseEvent.BUTTON1_DOWN_MASK);
-        pressSema.await();
+        if (!pressSema.await(2, TimeUnit.SECONDS)) {
+            throw new RuntimeException("Mouse is not pressed");
+        }
         System.out.println("# Pressed");
 
         r.mouseMove(sLoc.x + bSize.width / 2, sLoc.y + bSize.height * 2);
-        exitSema.await();
+        if (!exitSema.await(1, TimeUnit.SECONDS)) {
+            throw new RuntimeException("Mouse is not moved");
+        }
         System.out.println("# Exited");
 
         wheelSema = new CountDownLatch(1);
         r.mouseWheel(1);
-        wheelSema.await();
+        if (!wheelSema.await(1, TimeUnit.SECONDS)) {
+            throw new RuntimeException("Mouse is not wheeled");
+        }
         System.out.println("# Wheeled 1");
 
         wheelSema = new CountDownLatch(1);
         r.mouseWheel(-1);
-        wheelSema.await();
+        if (!wheelSema.await(1, TimeUnit.SECONDS)) {
+            throw new RuntimeException("Mouse is not wheeled 2");
+        }
         System.out.println("# Wheeled 2");
 
         r.mouseRelease(MouseEvent.BUTTON1_DOWN_MASK);
-        releaseSema.await();
+        if (!releaseSema.await(1, TimeUnit.SECONDS)) {
+            throw new RuntimeException("Mouse is not released");
+        }
         System.out.println("# Released!");
     }
 
     public static void main(String[] args) throws Exception {
         WheelModifier test = new WheelModifier();
 
-        SwingUtilities.invokeAndWait(() -> test.createGui());
-        test.run();
+        try {
+            SwingUtilities.invokeAndWait(test::createGui);
+            test.run();
+        } finally {
+            SwingUtilities.invokeAndWait(test.f::dispose);
```

Comment:
It would be good to ensure that f is not NULL before dispose.

Reply:
> It would be good to ensure that f is not NULL before dispose.

I know we're doing it all the time… But is it really relevant?

This has been bothering me for a while now… Let's clear things up.

If the frame is `null`, something has gone awry. If the test code itself didn't throw the exception, we'll get:

```text
Exception in thread "main" java.lang.NullPointerException
        at WheelModifier.main(WheelModifier.java:173)
```

The bad thing is that any exception thrown from the test itself — in the `try` block — gets overridden by the exception thrown from the `finally` block. Thus, we get the same exception as above.

The overall result is that the test fails, yet it may hide the real reason, which makes it harder to analyse the root cause of the failure.

Taking into account the above, *`null`-check is good even though it makes the code longer.*

Reply:
Yes, if we have a null frame, I'd like to think that there's already an exception thrown by a well written test,
and a finally block isn't normally the place you'd want to have an accidental exception generated.